### ISSUE TITLE
Compile error in oref0-setup.sh when choosing to compile go code from source (reoccurence)

### ISSUE THREAD

**Issue by Deweyoxberg** [openaps/oref0#1334]
Started installation with following parameters, on a Pi HAT:
/root/src/oref0/bin/oref0-setup.sh --dir=/root/myopenaps --serial=redacted --cgm=G4-Go --bleserial=y --ns-host=https://redacted.herokuapp.com --api-secret=redacted --tty=/dev/spidev0.0 --max_iob=24 --enable=' autotune ' --radio_locale='WW' --btmac='redacted' --hardwaretype='explorer-hat' --radiotags='cc111x'

When the script starts retrieving packages for Go, it fails with the attached screenshot. There was a previous bug as outlined by 1063, which was apparently fixed in 0.7.0-dev. Error is occurring in 0.7.0 master.
![error](https://user-images.githubusercontent.com/417795/70381105-0a338900-1902-11ea-86f5-f0accfd493bb.png)

Previous bug report is here: https://github.com/openaps/oref0/issues/1063

**Setup Information (please complete the following information):**
G4-Go
0.7.0-master

**Comment by Deweyoxberg** [comment]
go version go1.12.5 linux/arm

**Comment by Deweyoxberg** [comment]
Manage to progress past the uint errors, with a import cycle now:

get "golang.org/x/sys/unix": verifying non-authoritative meta tag
Fetching https://golang.org/x/sys?go-get=1
Parsing meta tags from https://golang.org/x/sys?go-get=1 (status code 200)
golang.org/x/sys (download)
github.com/ecc1/radio (download)
github.com/ecc1/spi (download)
github.com/ecc1/nightscout (download)
github.com/ecc1/papertrail (download)
import cycle not allowed
package github.com/ecc1/medtronic
        imports bytes
        imports errors
        imports runtime
        imports internal/bytealg
        imports internal/cpu
        imports runtime
Couldn't go get medtronic

**Comment by scottleibrand** [comment]
Probably best to start deleting go stuff and re-running setup.

**Comment by Deweyoxberg** [comment]
> 
> 
> Probably best to start deleting go stuff and re-running setup.

Bit of a noob; any series of commands we need to execute to wipe Go from the Pi and retry it? I tried updating to most recent version but it falls back to 1.7.

**Comment by scottleibrand** [comment]
Try `rm -rf /root/go`. Or just reflash.

**Comment by Deweyoxberg** [comment]
Yeah the repo seems to be the root error here. Fully removed go and reinstalled with the install script, and still nada. Given other users are having similar issues, and given that the latest Go version is 13 and up, it would seem this isn't an APS issue but a Go issue. Thoughts?

**Comment by Deweyoxberg** [comment]
Just recalled that GCloud has been having issues all afternoon; could be residual errors from that outage. Will retry tomorrow, and update this report accordingly.

**Comment by scottleibrand** [comment]
Yeah, sounds like a google cloud problem. Per @cluckj: “google cloud stuff has been broken all afternoon”

**Comment by cluckj** [comment]
Now that google cloud is reliable again, this looks to be the result of an upstream bug: https://github.com/golang/go/issues/36032

**Comment by cluckj** [comment]
A temporary fix for this is to revert the commit (see below), but per @ecc1 there will likely be merge conflict issues once this is repaired upstream. (Which you can fix by deleting the `$GOPATH/src/golang.org/x/sys/unix` directory.)

```
cd $GOPATH/src/golang.org/x/sys/unix
git revert bc7efcf
```

**Comment by cluckj** [comment]
The bug is resolved upstream, I was just able to install without this issue.
